Sources/KeyboardSupport/KeyboardRespondable.swift

```diff
@@ -137,33 +137,48 @@ public extension KeyboardScrollable where Self: UIViewController {
     }
     
     private func adjustViewForKeyboardAppearance(with keyboardInfo: KeyboardInfo, firstResponder: UIView) {
+        guard let scrollView = keyboardScrollableScrollView else { return }
+        
         var mutableInset: UIEdgeInsets
-        if shouldPreserveContentInsetWhenKeyboardVisible, let originalContentInset = keyboardScrollableScrollView?.originalContentInset {
+        if shouldPreserveContentInsetWhenKeyboardVisible, let originalContentInset = scrollView.originalContentInset {
             mutableInset = originalContentInset
         } else {
             mutableInset = .zero
         }
         
         // Adjust scroll view insets for keyboard height
-        let keyboardSize = keyboardInfo.finalFrame.size
+        let keyboardHeight = keyboardInfo.finalFrame.height
         if #available(iOS 11.0, *) {
-            mutableInset.bottom += keyboardSize.height - view.safeAreaInsets.bottom
+            mutableInset.bottom += keyboardHeight - view.safeAreaInsets.bottom
         } else {
-            mutableInset.bottom += keyboardSize.height
+            mutableInset.bottom += keyboardHeight
         }
+        
         adjustScrollViewInset(mutableInset)
         
         // If active text field is hidden by keyboard, scroll so it's visible
-        let keyboardMinY = view.bounds.height - keyboardSize.height
-        
-        let firstResponderConvertedFrame = firstResponder.convert(firstResponder.bounds, to: nil)
-        let firstResponderMaxY = firstResponderConvertedFrame.maxY
-        
-        if firstResponderMaxY > keyboardMinY {
-            keyboardScrollableScrollView?.scrollRectToVisible(firstResponder.frame, animated: true)
+        if let textView = firstResponder as? UITextView {
+            // Get the frame of the cursor/selection to improve scrolling position for UITextView's
+            // DispatchQueue.async() is necessary because the selectedTextRange typically hasn't not been updated when UIResponder.keyboardWillShowNotification is posted
+            DispatchQueue.main.async {
```

Comment:
Feels a little weird, but I don't see any other way around it. Feels like something that should be Radar'ed :).